src/server/internalServer.cpp

```diff
@@ -132,6 +132,56 @@ std::vector<T> subrange(const std::vector<T>& v, size_t s, size_t n)
 
 } // unnamed namespace
 
+std::set<std::string> InternalServer::selectBooks(const RequestContext& request, const std::string& prefix) const
+{
+  // Try old API
+  try {
+    auto bookName = request.get_argument("content");
+    try {
+      return {mp_nameMapper->getIdForName(bookName)};
+    } catch (const std::out_of_range&) {
+      throw std::invalid_argument("The requested book doesn't exist.");
+    }
+  } catch(const std::out_of_range&) {
+    // We've catch the out_of_range of get_argument
+    // continue
+  }
+
+  // Does user directly gives us ids ?
+  try {
+    auto id_vec = request.get_arguments(prefix+"id");
+    if (id_vec.empty()) {
+      throw std::invalid_argument("You must provide a value for the id.");
+    }
+    return std::set<std::string>(id_vec.begin(), id_vec.end());
+  } catch(const std::out_of_range&) {}
+
+  // Use the names
+  try {
+    auto name_vec = request.get_arguments(prefix+"name");
+    if (name_vec.empty()) {
+      throw std::invalid_argument("You must provide a value for the name.");
+    }
+    std::set<std::string> bookIds;
+    for(const auto& bookName: name_vec) {
+      try {
+        bookIds.insert(mp_nameMapper->getIdForName(bookName));
+      } catch(const std::out_of_range&) {
+        throw std::invalid_argument("The requested book doesn't exist.");
```

Comment:
Which book doesn't exist?

Reply:
This is fixed in later commit with the `Error` class